AirCasting/SessionViews/ChartView.swift

```diff
@@ -52,26 +52,30 @@ class UI_PollutionChart: UIView {
 }
 
 struct ChartView: UIViewRepresentable {
-    
-    @ObservedObject var stream: MeasurementStreamEntity
-    var thresholds: [SensorThreshold]
+    private let stream: MeasurementStreamEntity
+    private let thresholds: [SensorThreshold]
 
+    @StateObject var chartCreator: ChartViewModel
```

Comment:
We shouldn't be using @StateObject like this, it's a source of truth and should be created in one place. 

1. SwiftBySundell: 
"Although it’s technically possible to only use StateObject from now on — I still recommend using ObservedObject when observing external objects, and to **only use StateObject when dealing with objects that are owned by a view itself**".  
https://www.swiftbysundell.com/articles/swiftui-state-management-guide/

2. Apple's developer comment on a similar thing: 
A @StateObject is only meant for instantiating an object at some point in your view hierarchy, as you've done with presenterStateWorking. This attribute tells SwiftUI to create the object only once, regardless of how many times it recreates the enclosing view. **It isn't meant for passing an object through the interface**, as you've done with presenterStateWorking(I think he meant presenterStateNotWorking). For that, use @ObservedObject, as you have with presenterObserved."
https://developer.apple.com/forums/thread/650618

Adding this quotes because we already had this conversation with Paweł and I wanted to support my words by some experts opinions ;D

Reply:
I know I already reviewed&accepted this PR and didn't add this comment at that time. I missed this line of code and I've just remembered you talked about @StateObject and wanted to take a look at how it's done :)

Reply:
This chartViewModel is very tightly associated with chartView (i.e, it's creating an array of [ChartDataEntries]. The logic is much better when we are creating it inside the view (and it belongs just to this view and is not passed through interface, which would be a problem according to quote 2). We can say that this class is owned by the view itself. In the current setup it is created only in one place. So I'm not convinced by those quotes that we should change that. However, we can discuss it further. Let's see what @pawelgil thinks

Reply:
https://www.google.com/url?sa=i&url=https%3A%2F%2Ftenor.com%2Fview%2Fthumbs-up-nice-friends-joey-tribbiani-matt-le-blanc-gif-15167765&psig=AOvVaw3NOWy7bQ0sIh0iApcCLUqh&ust=1633595830759000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCPig8eywtfMCFQAAAAAdAAAAABBE

Reply:
![image](https://user-images.githubusercontent.com/46443841/136169265-ce62adb5-9b2d-487b-bfeb-130640c4bd8b.png)